File: packages/actions/src/email-link/civ-email-link.ts
import { html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { CivBaseElement, LightDomTextMixin } from '@civui/core';
import '../link/civ-link.js';

/**
 * CivUI Email Link
 *
 * A clickable email address link with `mailto:` protocol and mail icon.
 *
 * @element civ-email-link
 *
 * @prop {string} address - The email address
 * @prop {string} label - Override display text (defaults to the address)
 * @prop {string} subject - Pre-filled email subject line
 *
 * @example
 * ```html
 * <civ-email-link address="help@va.gov"></civ-email-link>
 * <civ-email-link address="help@va.gov" label="Email us" subject="Benefits question"></civ-email-link>
 * ```
 */
@customElement('civ-email-link')
export class CivEmailLink extends LightDomTextMixin(CivBaseElement) {
  @property({ type: String }) address = '';
  @property({ type: String }) label = '';
  @property({ type: String }) subject = '';
  @property({ type: Boolean, reflect: true }) disabled = false;

  private get _mailtoHref(): string {
    if (!this.address) return '';
    const params = this.subject ? `?subject=${encodeURIComponent(this.subject)}` : '';
    return `mailto:${this.address}${params}`;
  }

  override render() {
    const text = this.label || this.address;

    return html`
      <civ-link
        label="${text}"
        href="${this._mailtoHref}"
        icon-start="mail"
        ?disabled="${this.disabled}"
      ></civ-link>
    `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'civ-email-link': CivEmailLink;
  }
}

File: packages/ios/Sources/CivUI/CivEmailLink.swift
// CivUI — CivEmailLink for SwiftUI
// Accessible email link following government design system patterns.

import SwiftUI

/// Accessible email link for government applications.
///
/// Renders a mailto link with optional subject line.
/// Mirrors the web `civ-email-link` component.
///
/// Usage:
/// ```swift
/// CivEmailLink(
///     address: "help@agency.gov",
///     label: "Email support",
///     subject: "Benefits question"
/// )
/// ```
public struct CivEmailLink: View {
    // MARK: - Properties

    /// Email address.
    public var address: String

    /// Link text. Defaults to the email address if empty.
    public var label: String

    /// Pre-filled subject line.
    public var subject: String

    /// Whether the link is disabled.
    public var isDisabled: Bool

    // MARK: - Internal State

    @Environment(\.colorScheme) private var colorScheme

    // MARK: - Initializer

    public init(
        address: String = "",
        label: String = "",
        subject: String = "",
        isDisabled: Bool = false
    ) {
        self.address = address
        self.label = label
        self.subject = subject
        self.isDisabled = isDisabled
    }

    // MARK: - Body

    public var body: some View {
        EmptyView()
    }

    // MARK: - Color Helper

    private func adaptiveColor(light: Color, dark: Color) -> Color {
        colorScheme == .dark ? dark : light
    }
}

// MARK: - Preview

#if DEBUG
struct CivEmailLink_Previews: PreviewProvider {
    static var previews: some View {
        CivEmailLink(
            address: "help@agency.gov",
            label: "Email support",
            subject: "Benefits question"
        )
        .padding()
    }
}
#endif
